MapboxNavigation/RouteMapViewController.swift

```diff
@@ -396,13 +396,30 @@ class RouteMapViewController: UIViewController {
         let instructionBannerHeight = topBannerContainerView.bounds.height
         let bottomBannerHeight = bottomBannerContainerView.bounds.height
         
-        var insets = UIEdgeInsets(top: instructionBannerHeight, left: 0.0, bottom: bottomBannerHeight, right: 0.0)
+        // Inset by the safe area to avoid notches.
+        var insets = mapView.safeArea
+        insets.top += instructionBannerHeight
+        insets.bottom += bottomBannerHeight
         
         if overviewing {
             insets += NavigationMapView.courseViewMinimumInsets
             
             let routeLineWidths = MBRouteLineWidthByZoomLevel.compactMap { $0.value.constantValue as? Int }
             insets += UIEdgeInsets(floatLiteral: Double(routeLineWidths.max() ?? 0))
+        } else if mapView.tracksUserCourse {
```

Comment:
RouteMapViewController is used by NavigationViewController but not CarPlayNavigationViewController. Developers are also responsible for implementing their own preview map or free-driving map using a standalone NavigationMapView. So this code would need to be duplicated in CarPlayNavigationViewController and application code. It would be better to keep the code in NavigationMapView if possible.

Reply:
After https://github.com/mapbox/mapbox-navigation-ios/pull/2211/commits/98551ddcbf5934226702dc0ce710266d86f809ce

CarPlayNavigationViewController and NavigationViewController contentInset() functions are very close and seems feasible to extract common part to NavigationMapView.
As I don't have enough understanding of navigation-ios API surface. it is better if someone else from navigation-ios team continue on formulating the API, that is if the workarounds taken here are fine.

Reply:
We’re going to address duplication between CarPlayNavigationViewController and NavigationViewController overall in a separate PR ahead of v1.0.